c7n/mu.py

```diff
@@ -119,18 +120,27 @@ def create(self):
             arc_prefix = os.path.relpath(root, venv_lib_path)
             files = self.filter_files(files)
             for f in files:
-                self._zip_file.write(
-                    os.path.join(root, f),
-                    os.path.join(arc_prefix, f))
+                f_path = os.path.join(root, f)
+                dest_path = os.path.join(arc_prefix, f)
+                self.add_file(f_path, dest_path)
 
     def add_file(self, src, dest):
-        self._zip_file.write(src, dest)
+        info = zipfile.ZipInfo(dest)
+        timestamp = os.path.getmtime(src)
+        info.date_time = datetime.timetuple(
```

Comment:
I thought it would be nice to preserve the date, but since the archive will only ever be used on Lamba, it’s probably not worth the overhead. I’ll take that out.